Show a message if the viewer was blocked by the user

davidmz opened the pull request:
When the user A (blocked by the user B) browses the page of user B, show  the message: "Nothing to show here. Perhaps %screenName% has blocked you."

indeyets commented on src/components/user-feed.jsx:
```diff
@@ -3,21 +3,31 @@ import React from 'react';
-    )}
-  </div>
-);
+export default props => {
```
Add a component name, while you're changing the code. `const UserFeed = (props) => …`

indeyets commented on src/components/user-feed.jsx:
```diff
@@ -3,21 +3,31 @@ import React from 'react';
+  const {viewUser, location: {query}} = props;
+  const isBlocked = viewUser.blocked;
+  const isPrivate = viewUser.isPrivate === '1' && !viewUser.subscribed && !viewUser.isItMe;
+  const amIBlocked = viewUser.isPrivate === '0' && viewUser.statistics.posts !== '0' && props.visibleEntries.length === 0 && (!('offset' in query) || query.offset === '0');
```
shouldn't first part of condition be `!isPrivate` instead?

davidmz replied:
isPrivate is a string, '0' or '1'. So I think it is better to explicitly compare it with string.

indeyets commented on src/components/user-feed.jsx:
```diff
@@ -3,21 +3,31 @@ import React from 'react';
-  </div>
-);
+export default props => {
+  const {viewUser, location: {query}} = props;
```
you can move destructuring right into component-function declaration: `({ viewUser, location: { query } }) => …`

indeyets commented on src/components/user-feed.jsx:
```diff
@@ -3,21 +3,31 @@ import React from 'react';
+      {isBlocked ? (
+        <div className="box-body">
+          <p>You have blocked <b>{viewUser.screenName}</b>, so all of their posts and comments are invisible to you.</p>
+          <p><a onClick={()=>props.userActions.unban({username: viewUser.username, id: viewUser.id})}>Un-block</a></p>
```
inline declaration of handlers is a bad pattern. No need to fix this here, but we should figure out something in the future

see https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

davidmz committed:
Show a message if the viewer was blocked by the user

When the user A (blocked by the user B) browses the page of user B, show the message: "Nothing to show here. Perhaps %screenName% has blocked you."
